spec/ddtrace/contrib/sucker_punch/patcher_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'ddtrace/contrib/support/spec_helper'
+require 'sucker_punch'
+require 'ddtrace'
+require_relative 'dummy_worker'
+
+RSpec.describe 'Sinatra instrumentation' do
+  before do
+    Datadog.configure do |c|
+      c.use :sucker_punch
+    end
+
+    ::SuckerPunch::Queue.clear
+    ::SuckerPunch::RUNNING.make_true
+  end
+
+  after do
+    ::SuckerPunch::Queue.clear
+  end
+
+  around do |example|
+    # Reset before and after each example; don't allow global state to linger.
+    Datadog.registry[:sucker_punch].reset_configuration!
+    example.run
+    Datadog.registry[:sucker_punch].reset_configuration!
+  end
+
+  context 'successful job' do
+    subject(:dummy_work_success) { ::DummyWorker.perform_async }
```

Comment:
When moving these tests to RSpec, we can avoid adding classes to the global namespace by only stubbing them during the execution of the test.

I suggest we do that for `DummyWorker`.

If don't actually need a globally defined class: https://github.com/DataDog/dd-trace-rb/blob/61ef72b7df6b46061e4e00e0e278deb03a7683c5/spec/ddtrace/workers/async_spec.rb#L10-L12

If you really need a globally defined class: https://github.com/DataDog/dd-trace-rb/blob/64ec7458bd9e6fe010382cdd655044a8aca7264f/spec/ddtrace/contrib/sidekiq/server_tracer_spec.rb#L38-L46

Reply:
makes sense, i  tried to update this and then remove the `dummy_worker.rb` class file entirely and just have it as a stub.